dashboard/app/models/script.rb

```diff
@@ -1342,7 +1342,7 @@ def update_text(unit_params, unit_text, metadata_i18n, general_params)
           login_required: general_params[:login_required].nil? ? false : general_params[:login_required], # default false
           wrapup_video: general_params[:wrapup_video],
           family_name: general_params[:family_name].presence ? general_params[:family_name] : nil, # default nil
-          published_state: general_params[:published_state],
+          published_state: (unit_group.present? && general_params[:published_state] == unit_group.published_state) ? nil : general_params[:published_state],
```

Comment:
nice, I don't think I have a test on the "Hide this unit within this course" checkbox functionality, but this looks like it should preserve that.

Reply:
I did test that functionality locally and it still worked!